elasticapm/conf/__init__.py

```diff
@@ -101,6 +101,22 @@ def __set__(self, instance, value):
         instance._values[self.dict_key] = value
 
 
+class _DictConfigValue(_ConfigValue):
+    def __init__(self, dict_key, item_separator=",", keyval_separator="=", **kwargs):
+        self.item_separator = item_separator
+        self.keyval_separator = keyval_separator
+        super(_DictConfigValue, self).__init__(dict_key, **kwargs)
+
+    def __set__(self, instance, value):
+        if isinstance(value, compat.string_types):
+            items = (item.split(self.keyval_separator) for item in value.split(self.item_separator))
+            value = {key: self.type(val) for key, val in items}
+        elif not isinstance(value, dict):
+            # TODO: better error handling
+            value = None
+        instance._values[self.dict_key] = value
```

Comment:
should probably check that the tag keys match `TAG_RE` here? also, truncate value to 1024 chars?

Reply:
Both done in 06b64dc9d3ec